nipype/interfaces/base/tests/test_experimental.py

```diff
@@ -0,0 +1,60 @@
+"""Proof of concept."""
+from ....utils.filemanip import fname_presuffix
+from ... import base as nib
+from ..experimental import AutoOutputInterface
+
+_TOOL_OUTPUT = r"""\
+  _____________
+< Nipype rocks! >
+  =============
+                  \
+                   \
+                     ^__^
+                     (oo)\_______
+                     (__)\       )\/\
+                         ||----w |
+                         ||     ||
+
+
+"""
+
+
+def _parse_err_cb(in_text):
+    if '3.0' in in_text:
+        return 3.0
+    raise RuntimeError('Failed Parsing')
+
+
+class _InputSpec(nib.TraitedSpec):
+    foo = nib.traits.Int(desc="a random int")
+    moo = nib.traits.Int(desc="a random int", mandatory=False)
+    hoo = nib.traits.Int(desc="a random int", usedefault=True)
+    zoo = nib.File(desc="a file")
+    woo = nib.File(desc="a file")
+
+
+def test_AutoOutputInterface():
+    """Proof of concept."""
+    class _OutputSpec(nib.TraitedSpec):
+        out_woo = nib.File('{woo}_brain', usedefault=True)
+        out_std = nib.Str(stdout=True)
+        out_err = nib.traits.Float(stderr=_parse_err_cb)
```

Comment:
this functional form is the part that would need to be formalized if we are going to transform the interface into a specification for mapping between software (e.g., nipype1 to pydra spec).

Reply:
As I mentioned above/below, the std out/err is actually easier once we convert these into metadata. Parse callables can be reused right away and the semantics of the metadata are clear enough (this is basically one of the most appealing points of this suggestion).

``_list_outputs`` is going to be harder to convert from code to metadata.